hypershift-operator/controllers/hostedcluster/karpenter.go

```diff
@@ -65,71 +66,54 @@ spec:
 		return fmt.Errorf("failed to create configmap: %w", err)
 	}
 
-	// Managed a NodePool to generate userData for Karpenter instances
-	// TODO(alberto): consider invoking the token library to manage the karpenter userdata programmatically,
-	// instead of via NodePool API.
 	nodePool := &hyperv1.NodePool{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "karpenter",
 			Namespace: hcluster.Namespace,
 		},
-	}
-	spec := hyperv1.NodePoolSpec{
-		ClusterName: hcluster.Name,
-		Replicas:    ptr.To(int32(0)),
-		Release:     hcluster.Spec.Release,
-		Config: []corev1.LocalObjectReference{
-			{
-				Name: taintConfigName,
-			},
-		},
-		Management: hyperv1.NodePoolManagement{
-			UpgradeType: hyperv1.UpgradeTypeReplace,
-			Replace: &hyperv1.ReplaceUpgrade{
-				Strategy: hyperv1.UpgradeStrategyRollingUpdate,
-				RollingUpdate: &hyperv1.RollingUpdate{
-					MaxUnavailable: ptr.To(intstr.FromInt(0)),
-					MaxSurge:       ptr.To(intstr.FromInt(1)),
+		Spec: hyperv1.NodePoolSpec{
+			ClusterName: hcluster.Name,
+			Replicas:    ptr.To(int32(0)),
+			Release:     hcluster.Spec.Release,
+			Config: []corev1.LocalObjectReference{
+				{
+					Name: taintConfigName,
 				},
 			},
-			AutoRepair: false,
-		},
-		Platform: hyperv1.NodePoolPlatform{
-			Type: hyperv1.AWSPlatform,
-			AWS: &hyperv1.AWSNodePoolPlatform{
-				InstanceType: "m5.large",
-				Subnet: hyperv1.AWSResourceReference{
-					// TODO(alberto): this is just to pass cel.
-					// Setting an ID instead of filter would break publicAndPrivate topology because the AWSEndpointService won't find the subnet.
-					// We'll move to generate the userdata for karpenter programatically.
-					Filters: []hyperv1.Filter{
-						{
-							Name:   "subnet-none",
-							Values: []string{"none"},
-						},
+			Management: hyperv1.NodePoolManagement{
```

Comment:
we don't need to define .management, do we?

Reply:
removed